### ISSUE TITLE
Support Windows resource embedding

### ISSUE THREAD

**Issue by gspencergoog** [dart-lang/sdk#45344]
From @timsneath:
> Modern Windows apps (whether console or GUI) embed an XML-based application manifest to describe application metadata. The manifest includes basic signatures, high-DPI support information, OS compatibility levels, and heap API configuration. All mainstream Windows compilers and linkers, to the best of my knowledge, support the inclusion of a manifest, which is linked into the executable as a Windows resource file. Without this, Dart is prevented from accessing certain APIs including modern message boxes (TaskDialog), and various configuration options are set to a legacy Windows 7 compatible mode of execution.
> 
> The application manifest issue is a more specific case of resource embedding, which allows other binary and string resources to be statically linked into a Windows executable (including application icons, fonts, dialog resources and graphics). For most of these scenarios, Windows allows dynamic loading of resources at runtime, but manifests must be loaded prior to the app initialization.
> 
> The work itself should be fairly straightforward: dart compile would need an additional option to point to a resource, which would then be linked into the output of the compilation process.

We (the Flutter Desktop team) would like this ability to enable the possibility of rewriting our Runner code in Dart using FFI.

/cc @cbracken @clarkezone @csells

**Comment by vsmenon** [comment]
Is this request specifically for `dart compile exe`?

**Comment by gspencergoog** [comment]
Yes. (and I suppose any other dart compile targets for Windows, if there are any).
